### PR TITLE
Push sdist

### PR DESCRIPTION
**Pull request by mgeplf**
* dev versions of libsonata weren't being pushed to PyPI, allow that
* Note: wheels are still created in separate repo

### CODE REVIEW

**Comment by haleepfl** [comment]
Why not just tag master with a dev version and allow travis to handle the upload in the usual way?

**Comment by mgeplf** [comment]
> Why not just tag master with a dev version and allow travis to handle the upload in the usual way?

I'd prefer not to have to tag each commit; this way setuptools_scm uploads a `devX` version for each merged PR, which seems reasonable to me.

File: .travis.yml [hunk 1]
```diff
@@ -36,18 +36,15 @@ cache:
     - $HOME/.cache/pip
 
 before_install:
-  - rm -rf dist *.egg-info && python setup.py sdist
+    - rm -rf dist *.egg-info
 
 install: skip
```
Comment (wizmer) [inline .travis.yml:41]: Is this line useful ?

File: .travis.yml [hunk 2]
```diff
@@ -36,18 +36,15 @@ cache:
...
 script:
-  - ci/cpp_test.sh
-  - ci/python_test.sh
+    - ci/cpp_test.sh
+    - ci/python_test.sh
 
 deploy:
+  provider: script
+  script: ci/python_sdist.sh
```
Comment (wizmer) [inline .travis.yml:48]: I would use a filename that suggest this script has a side effect. sdist_to_pypi.sh or something like this.

File: ci/python_sdist.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# upload the sdist files to PyPI
+
+set -euo pipefail
+
+if [[ -z "$TRAVIS" ]]; then
+    echo "This script is meant to be run on travis-ci.com"
+    exit -1
+fi
+
+env
```
Comment (wizmer) [inline ci/python_sdist.sh:12]:
```suggestion
```

**Comment by haleepfl** [comment]
> > Why not just tag master with a dev version and allow travis to handle the upload in the usual way?
> 
> I'd prefer not to have to tag each commit; this way setuptools_scm uploads a `devX` version for each merged PR, which seems reasonable to me.

I see. Why not just replace `on: tags: true` with `on: branch: master` then?

https://docs.travis-ci.com/user/deployment#conditional-releases-with-on

**Pull request merged**